lib/kitchen/verifier/pester.rb

```diff
@@ -302,30 +418,55 @@ def really_wrap_shell_code(code)
         windows_os? ? really_wrap_windows_shell_code(code) : really_wrap_posix_shell_code(code)
       end
 
+      # Get the defined shell or fall back to pwsh, unless we're on windows where we use powershell
+      # call via sudo if sudo is true.
+      # This allows to use pwsh-preview instead of pwsh, or a full path to a specific binary.
+      def shell_cmd
+        if !config[:shell].nil?
+          config[:sudo] ? "sudo #{config[:shell]}" : "#{config[:shell]}"
+        elsif windows_os?
+          'powershell'
+        else
+          config[:sudo] ? "sudo pwsh" : "pwsh"
+        end
+      end
+
       def really_wrap_windows_shell_code(code)
-        wrap_shell_code(Util.outdent!(use_local_powershell_modules(code)))
+        my_command = <<-PWSH
+          echo "Running as '$(whoami)'..."
+          New-Item -ItemType Directory -Path '#{config[:root_path]}/modules' -Force -ErrorAction SilentlyContinue
+          Set-Location -Path "#{config[:root_path]}"
+          # Send the pwsh here string to the file kitchen_cmd.ps1
+          @'
+          Set-ExecutionPolicy Unrestricted -force
```

Comment:
I think this should be in a try/catch, similar to how the old code was (lines 377-344 in the "before" diff) -- it can throw an error if the policy can't be modified for some reason. That error would bomb this out and fail the verifier run, but isn't necessarily a failure state. The pre-existing execution policy might still work, even if this command can't change it.

Reply:
fixed